### PR TITLE
[ODS-4435] Merge the net core changes back into the main branch

### PR DESCRIPTION
**Pull request by jamessmckay**

### CODE REVIEW

File: Application/EdFi.Ods.Admin.Tests/EdFi.Ods.Admin.Tests.csproj [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>net48</TargetFramework>
+    <TargetFrameworks>net48</TargetFrameworks>
```
Comment (blmeyers) [inline Application/EdFi.Ods.Admin.Tests/EdFi.Ods.Admin.Tests.csproj:3]: Should this be dual targeting netcoreapp3.1?

File: Application/EdFi.Ods.Api/Controllers/DataManagementControllerBase.cs [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+
+        [CheckModelForNull]
+        [HttpPut("{id}")]
+        [ProducesResponseType(StatusCodes.Status201Created)]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status412PreconditionFailed)]
+        public virtual async Task<IActionResult> Put([FromBody] TPutRequest request, [FromQuery] Guid id)
+        {
+            // Manual binding of Id to main request model
```
Comment (blmeyers) [inline Application/EdFi.Ods.Api/Controllers/DataManagementControllerBase.cs:218]: We lost ModelState.IsValid checks here (as added in https://github.com/Ed-Fi-Alliance/Ed-Fi-ODS/commit/845bae2ce7470e91a957222be4fd108b463b280d).  Is this unnecessary in .net core or are we regressing?

File: Application/EdFi.Common/EdFi.Common.csproj [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>netstandard2.0;net48</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0;netstandard2.1;net48</TargetFrameworks>
```
Comment (blmeyers) [inline Application/EdFi.Common/EdFi.Common.csproj:3]: Thought we were dropping to just 2.1?  Do we have a separate ticket on making this entire project only 2.1 (no 2.0 or net48).

File: Application/EdFi.Ods.Admin.Tests/EdFi.Ods.Admin.Tests.csproj (hunk 1 above)
Comment (jamessmckay) [reply Application/EdFi.Ods.Admin.Tests/EdFi.Ods.Admin.Tests.csproj:3]: we cant target yet until we have converted the base project. that is on a separate ticket. (ODS-3505)

File: Application/EdFi.Ods.Api/Controllers/DataManagementControllerBase.cs (hunk 2 above)
Comment (jamessmckay) [reply Application/EdFi.Ods.Api/Controllers/DataManagementControllerBase.cs:218]: looks like i missed that change. I will create a ticket. ODS-4494

File: Application/EdFi.Common/EdFi.Common.csproj (hunk 3 above)
Comment (jamessmckay) [reply Application/EdFi.Common/EdFi.Common.csproj:3]: We are but that is another ticket. ODS-4454

File: Application/EdFi.Ods.Features/Composites/Infrastructure/HqlBuilder.cs [hunk 4]
```diff
@@ -0,0 +1,1009 @@
...
+using EdFi.Ods.Common.Specifications;
+using EdFi.Ods.Common.Utils.Extensions;
+using log4net;
+using NHibernate;
+using NHibernate.Exceptions;
+using NHibernate.Transform;
+
+namespace EdFi.Ods.Features.Composites.Infrastructure
+{
+        public class HqlBuilder : ICompositeItemBuilder<HqlBuilderContext, CompositeQuery>
```
Comment (blmeyers) [inline Application/EdFi.Ods.Features/Composites/Infrastructure/HqlBuilder.cs:29]:
```suggestion
    public class HqlBuilder : ICompositeItemBuilder<HqlBuilderContext, CompositeQuery>
```

File: tests/EdFi.Admin.DataAccess.UnitTests/EdFi.Admin.DataAccess.UnitTests.csproj [hunk 5]
```diff
@@ -17,16 +17,21 @@
...
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
     <PackageReference Include="Shouldly" Version="3.0.2" />
+    <PackageReference Include="TeamCity.VSTest.TestAdapter" Version="1.0.22" />
   </ItemGroup>
   <ItemGroup>
     <ProjectReference Include="..\..\Application\EdFi.Admin.DataAccess\EdFi.Admin.DataAccess.csproj" />
+    <ProjectReference Include="..\..\Application\EdFi.Common\EdFi.Common.csproj" />
+    <ProjectReference Include="..\..\Application\EdFi.Ods.Common\EdFi.Ods.Common.csproj" />
```
Comment (blmeyers) [inline tests/EdFi.Admin.DataAccess.UnitTests/EdFi.Admin.DataAccess.UnitTests.csproj:35]: We just had a ticket to remove references to this from EdFi.Admin.DataAccess, why are we adding it as a reference for the tests?

File: tests/EdFi.Admin.DataAccess.UnitTests/EdFi.Admin.DataAccess.UnitTests.csproj [hunk 6]
```diff
@@ -17,16 +17,21 @@
...
+    <PackageReference Include="NUnit3TestAdapter" Version="3.17.0">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
     <PackageReference Include="Shouldly" Version="3.0.2" />
+    <PackageReference Include="TeamCity.VSTest.TestAdapter" Version="1.0.22" />
   </ItemGroup>
   <ItemGroup>
     <ProjectReference Include="..\..\Application\EdFi.Admin.DataAccess\EdFi.Admin.DataAccess.csproj" />
+    <ProjectReference Include="..\..\Application\EdFi.Common\EdFi.Common.csproj" />
```
Comment (blmeyers) [inline tests/EdFi.Admin.DataAccess.UnitTests/EdFi.Admin.DataAccess.UnitTests.csproj:34]: Should this be a package reference like the project its testing?  Do we even need it?

Comment (jamessmckay) [reply tests/EdFi.Admin.DataAccess.UnitTests/EdFi.Admin.DataAccess.UnitTests.csproj:34]: this should be testing the project imho it is specific to the that class. not sure what value these unit test give but it was setup by another developer

Comment (blmeyers) [reply tests/EdFi.Admin.DataAccess.UnitTests/EdFi.Admin.DataAccess.UnitTests.csproj:34]: A test project for testing a library should directly reference the project, but shouldn't be using project references for the library's package dependencies.  Too much risk of getting misaligned there.  It should be package references everywhere.

**Comment by blmeyers** [comment]
Note, this PR was brought in with a merge commit on purpose, as per discussion in team meetings.  Due to the complicated nature of this change it was requested to keep the underlying history of the original branch.

**Pull request merged**